### PR TITLE
changed f64 to u64 for navigation start timing until it had to be float

### LINKED ISSUE #18997: Figure out why navigation_start_precise and paint timing metrics are f64
As far as I can tell, all of the code that originally provides a value for these matrics casts a u64 to an f64, and I cannot figure out why.

### PR DESCRIPTION
**Pull request by avadacatavra**
---

- [X] `./mach build -d` does not report any errors
- [X] `./mach test-tidy` does not report any errors
- [X] These changes fix #18997 (github issue number if applicable).

- [ ] There are tests for these changes OR
- [ ] These changes do not require tests because _____

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19077)

### CODE REVIEW

**Comment by avadacatavra** [comment]
@jdm r?

File: components/script/dom/performance.rs [hunk 1]
```diff
@@ -120,7 +120,8 @@ pub struct Performance {
 impl Performance {
     fn new_inherited(global: &GlobalScope,
                      navigation_start: u64,
-                     navigation_start_precise: f64) -> Performance {
+                     navigation_start_precise: u64) -> Performance {
+        let navigation_start_precise = navigation_start_precise as f64;
```
Comment (jdm) [inline components/script/dom/performance.rs]: Let's change PerformanceTiming to accept a u64 as well, and make PerformanceTiming::navigation_start_precise() return u64. We can simplify Performace::now() to only cast the result of the calculation instead.

**Review by jdm — CHANGES_REQUESTED** [review]

**Comment by avadacatavra** [comment]
...oops i messed up my rebase

File: components/script/dom/performance.rs [hunk 2]
```diff
@@ -114,13 +114,14 @@ pub struct Performance {
...
-    navigation_start_precise: f64,
+    navigation_start_precise: u64,
 }
 
 impl Performance {
     fn new_inherited(global: &GlobalScope,
                      navigation_start: u64,
-                     navigation_start_precise: f64) -> Performance {
+                     navigation_start_precise: u64) -> Performance {
+        let navigation_start_precise = navigation_start_precise;
```
Comment (jdm) [inline components/script/dom/performance.rs]: This isn't necessary now.

File: components/script/dom/performance.rs [hunk 3]
```diff
@@ -260,7 +261,7 @@ impl Performance {
             Some(ref timing) => timing.navigation_start_precise(),
             None => self.navigation_start_precise,
         };
-        (time::precise_time_ns() as f64 - nav_start) / 1000000 as f64
+        (time::precise_time_ns() as f64 - nav_start as f64) / 1000000 as f64
```
Comment (jdm) [inline components/script/dom/performance.rs]: We should be able to write this as `(time::precise_time_ns() - nav_start) as f64 / 1000000.` now.

**Review by jdm — CHANGES_REQUESTED** [review]

**Comment by jdm** [comment]
@bors-servo: r+
That is much nicer to read!

**Pull request merged**